### PR TITLE
fix: return always a full projection in multilayerAxisEntry

### LINKED ISSUE #1592: `maxLabelBboxHeight` can be undefined
A kibana user experienced an error where the server-side rendering of a dashboard in PDF the following error in the console appeared:  

```
Cannot read properties of undefined (reading 'maxLabelBboxHeight')
```

~We should find the possible root cause of this undefined value causing a blank chart rendered~

As notified in https://github.com/elastic/kibana/issues/134025 the issue appears when resizing a multi-layer time axis to a small size, causing the error to surface

### PR DESCRIPTION
**Pull request by markov00**
## Summary

When resizing too much a time series chart with a multilayer time axis, the function that computes the ticks can receive a a zero-length array of `layers`.  Typescript probably doesn't pick up correctly a mis-typed reduce and it propagated through without getting noticed causing an un-recoverable error as reported in the [linked issue](#https://github.com/elastic/elastic-charts/issues/1592) and the Kibana one.

This PR fixes it by adding the correct return type to the function and fixing the reduce function initialization.

The timeslip story is also updated with a switch to show the legend simplifying the reproducibility of the issue.

## Issues

fix #1592

### Checklist

- [x] The proper **chart type** label has been added (e.g. `:xy`, `:partition`)
- [x] The proper **feature** labels have been added (e.g. `:interactions`, `:axis`)
- [x] All related issues have been linked (i.e. `closes #123`, `fixes #123`)
- [ ] Unit tests have been added or updated to match the most common scenarios
- [ ] The proper documentation and/or storybook story has been added or updated

### CODE REVIEW

File: packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts [hunk 1]
```diff
@@ -63,8 +62,8 @@ export function multilayerAxisEntry(
       fallbackAskedTickCount: NaN,
     };
   };
-  return layers.reduce(
-    (combinedEntry: { ticks: AxisTick[] }, l: TimeRaster<TimeBin>, detailedLayerIndex) => {
+  return layers.reduce<Projection>(
```
Comment (monfera) [inline packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts:65]: Nice, totally worth the couple of added lines for the initial value

File: packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts [hunk 2]
```diff
@@ -85,8 +84,9 @@ export function multilayerAxisEntry(
       }
 
       return {
-        ...entry,
         ...combinedEntry,
+        ...entry,
+        scale,
```
Comment (monfera) [inline packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts:89]: `entry` has the `scale` already (OK to keep it)

**Review by monfera — APPROVED** [review]

File: packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts [hunk 3]
```diff
@@ -97,6 +97,16 @@ export function multilayerAxisEntry(
...
+      ticks: [],
+      labelBox: {
+        maxLabelBboxWidth: 0,
+        maxLabelBboxHeight: 0,
+        maxLabelTextWidth: 0,
+        maxLabelTextHeight: 0,
+        isHidden: true,
+      },
+      scale,
+    }, // this should turn into a full Projection
```
[the remark this answers is not in the document]
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts:110]: :D yes

File: packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts (hunk 2 above)
Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts:89]: true, and also combinedEntry :)

Comment (markov00) [reply packages/charts/src/chart_types/xy_chart/axes/timeslip/multilayer_ticks.ts]: done https://github.com/elastic/elastic-charts/pull/1710/commits/ba7bddfb309e30216358bedf69638ef2ee5911f6

**Pull request merged**